### PR TITLE
feat(eslint-plugin-lit-a11y): interactive-supports-focus

### PR DESCRIPTION
**Pull request by marcoswebdev**
Initial structure for https://github.com/open-wc/open-wc/pull/1781 interactive-supports-focus rule

Example: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/interactive-supports-focus.md

### CODE REVIEW

**Comment by thepassle** [comment]
Nice, this is great!

Would it be possible to add a few more test cases, and remove the unused/commented testcases? 🙏

File: packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/**
+ * @fileoverview
+ * @author open-wc
+ */
+
+const includes = require('array-includes');
```
Comment (bennypowers) [inline packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js:6]: We don't need this shim, 
https://node.green/#ES2016-features-Array-prototype-includes

so let's just use `Array#includes` instead, like we're doing here on L#63

File: packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    const interactiveProps = [...eventHandlersByType.mouse, ...eventHandlersByType.keyboard];
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      TaggedTemplateExpression: node => {
+        if (
```
Comment (bennypowers) [inline packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js:51]: We have a helper in utils, so you can
```js
const { isHtmlTaggedTemplate } = require('../utils/isLitHtmlTemplate.js');
```
and here:
```suggestion
        if (isHtmlTaggedTemplate(node)) {
```

File: packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                // && !isNonInteractiveElement(element.name, attributes)
+                // && !isNonInteractiveRole(element.name, attributes)
+                !hasTabindex
+              ) {
+                const role = getAttrVal(element.attribs.role);
+                if (includes(tabbable, role)) {
+                  // Always tabbable, tabIndex = 0
+                  context.report({
+                    node,
+                    message: `Elements with the '${role}' interactive role must be tabbable.`,
```
Comment (bennypowers) [inline packages/eslint-plugin-lit-a11y/lib/rules/interactive-supports-focus.js:92]: eslint has it's own interpolation feature which makes it easier to reuse messages.

this file is a good use case for messageId as well

take a look at the eslint docs:

https://eslint.org/docs/developer-guide/working-with-rules#using-message-placeholders

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Comment by marcoswebdev** [comment]
I have created a new PR where I fix your suggestions @bennypowers!

**Pull request closed without merging**